paywall/src/data-iframe/blockchainHandler/BlockchainHandler.ts

```diff
@@ -258,6 +283,10 @@ export default class BlockchainHandler {
     this.web3Service.on('key.updated', (_: any, key: KeyResult) => {
       key.lock = normalizeLockAddress(key.lock)
       this.store.keys[key.lock] = key
+      this.sendDataWhenKeyExpires(key.expiration)
+      // note: if this next line is ever removed, it will need to be
+      // added inside sendDataWhenKeyExpires() to ensure expired
```

Comment:
It looks like this is actually called inside of `sendDataWhenKeyExpires` already?

Reply:
line 232 would need to have the `dispatchChangesToPostOffice`. The flow is:

1. if the key is already expired, the `setTimeout` line is not called, and the dispatch is handled by line `290`.
2. if the key is not expired, the `setTimeout` sets up the dispatch.

The note is just to ensure the dependency on the dispatch existing in `'key.updated'` is explicit, so if in a future refactor that line is removed, we don't lose the "expired" state.